### PR TITLE
Fix tidy tripping up on  untracked files with special characters in their name

### PR DESCRIPTION
**Pull request by majaha**
Previously, the tidy tool would fault if an untracked file had a space or other special characters in its name. If there was an untracked file "foo bar", it would include the quoting in it's path and split on the first space, giving output like this:
`skip untracked path "foo during rustfmt invocations`

### CODE REVIEW

File: src/bootstrap/src/core/build_steps/format.rs [hunk 1]
```diff
@@ -142,14 +142,17 @@ pub fn format(build: &Builder<'_>, check: bool, paths: &[PathBuf]) {
...
+                build
+                    .config
+                    .git()
+                    .arg("status")
+                    .arg("--porcelain")
+                    .arg("-z")
+                    .arg("--untracked-files=normal"),
+            );
+            let untracked_paths = untracked_paths_output.split_terminator('\0').filter_map(
+                |entry| entry.strip_prefix("?? "), /* returns None if the prefix doesn't match */
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/build_steps/format.rs:154]: a nit suggestion:

```suggestion
                |entry| entry.strip_prefix("?? "), // returns None if the prefix doesn't match
```

**Review by onur-ozkan — APPROVED** [review]
LGTM other than 1 suggestion

File: src/bootstrap/src/core/build_steps/format.rs (hunk 1 above)
Comment (majaha) [reply src/bootstrap/src/core/build_steps/format.rs:154]: Looks good, I missed that when the formatter split it over a few lines.

**Comment by onur-ozkan** [comment]
Thanks for the fix. Can you squash the commits?

**Comment by majaha** [comment]
Ah sorry, I just saw this. Should be ready now!

**Pull request merged**